### PR TITLE
Roe 2309 remove save resume flag instances

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-2309

### Change description
Flag and associated code removed, test updated, details on the ticket linked above.

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/update/update.resume.submission.controller.ts [hunk 1]
```diff
@@ -1,10 +1,8 @@
 import { NextFunction, Request, Response } from "express";
 
 import * as config from "../../config";
-
-import { isActiveFeature } from "../../utils/feature.flag";
 import { getResumePage } from "../shared/common.resume.submission.controller";
 
 export const get = (req: Request, res: Response, next: NextFunction) => {
-  getResumePage(req, res, next, isActiveFeature(config.FEATURE_FLAG_ENABLE_UPDATE_SAVE_AND_RESUME), config.SECURE_UPDATE_FILTER_URL);
+  getResumePage(req, res, next, config.SECURE_UPDATE_FILTER_URL);
```
[the remark this answers is not in the document]
Comment (mattch1) [reply src/controllers/update/update.resume.submission.controller.ts:7]: accident

[the remark this answers is not in the document]
Comment (mattch1) [reply src/controllers/update/update.resume.submission.controller.ts:7]: I remember why it calls the same function as resume submission controller.

[the remark this answers is not in the document]
Comment (markpit) [reply src/controllers/update/update.resume.submission.controller.ts:7]: looks like we'll need to revert that resume submission controller then, as with the changes the update save and resume flag is effectively set to true. I would revert the resume submission controller and just pass "true" in from the registration controller, then the update controller can continue to use the flag.

File: test/controllers/payment.controller.spec.ts [hunk 2]
```diff
@@ -68,18 +67,7 @@ describe("Payment controller", () => {
...
-    expect(resp.status).toEqual(302);
-    expect(resp.text).toEqual(`${FOUND_REDIRECT_TO} ${CHECK_YOUR_ANSWERS_URL}`);
-    expect(mockLoggerDebugRequest).toHaveBeenCalledTimes(1);
-    expect(mockLoggerInfoRequest).toHaveBeenCalledTimes(1);
-    expect(mockCreateAndLogErrorRequest).not.toHaveBeenCalled();
-  });
-
-  test(`should redirect to ${PAYMENT_FAILED_PAGE} page, Payment failed somehow and feature flag active`, async () => {
+  test(`should redirect to ${PAYMENT_FAILED_PAGE} page, Payment failed somehow`, async () => {
     mockIsActiveFeature.mockReturnValueOnce(true);
```
Comment (markpit) [inline test/controllers/payment.controller.spec.ts:71]: is the mockIsActiveFeature.mockReturnValueOnce(true); still needed on line 71? if we've removed the feature flag check in the payment controller then we might not need to set the mockIsActiveFeature

File: test/controllers/sign.out.controller.spec.ts [hunk 3]
```diff
@@ -42,7 +42,7 @@ describe("Sign Out controller", () => {
   });
 
   describe("GET tests", () => {
-    test(`renders the ${config.SIGN_OUT_PAGE} page, when FEATURE_FLAG_ENABLE_SAVE_AND_RESUME_17102022 is active, with ${config.MANAGING_OFFICER_CORPORATE_PAGE} as back link`, async () => {
+    test(`renders the ${config.SIGN_OUT_PAGE} page, with ${config.MANAGING_OFFICER_CORPORATE_PAGE} as back link`, async () => {
```
Comment (markpit) [inline test/controllers/sign.out.controller.spec.ts:45]: might be able to remove the mockIsActiveFeature.mockReturnValueOnce(true) here too. If it works, probably good idea to check the other tests that have changed and see if the mockIsActiveFeature can be removed in those too.

File: test/controllers/update/sign.out.controller.spec.ts [hunk 4]
```diff
@@ -36,7 +36,7 @@ describe("SIGN OUT controller", () => {
     jest.clearAllMocks();
   });
   describe("GET tests", () => {
-    test(`renders the ${config.SIGN_OUT_PAGE} page, when FEATURE_FLAG_ENABLE_UPDATE_SAVE_AND_RESUME is active,  with ${config.SECURE_UPDATE_FILTER_PAGE} as back link`, async () => {
+    test(`renders the ${config.SIGN_OUT_PAGE} page, with ${config.SECURE_UPDATE_FILTER_PAGE} as back link`, async () => {
```
Comment (markpit) [inline test/controllers/update/sign.out.controller.spec.ts:39]: i think this is the wrong test file being updated. Doesn't look like we have changed controllers/update/update.sign.out.controller.ts

File: test/utils/save.and.continue.spec.ts [hunk 5]
```diff
@@ -23,15 +20,7 @@ describe('saveAndContinue test suite', () => {
   });
 
   test('should call updateOverseasEntity if feature flag is true', async () => {
```
Comment (markpit) [inline test/utils/save.and.continue.spec.ts:22]: utils/save.and.continue.ts still has the update flag in it so we will need to keep the tests for update flag = true and update flag is false.

File: src/controllers/update/update.resume.submission.controller.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mwestacott) [reply src/controllers/update/update.resume.submission.controller.ts]: Not sure why there is an update save and resume flag anyway, as that's going into LIVE _with_ Update. So really, just the Update feature flag would have been enough.

File: views/sign-out.html [hunk 6]
```diff
@@ -16,12 +16,10 @@
   }) }}
 {% endblock %}
 
-{% if saveAndResume and (journey == "register") %}
+{% if journey == "register" %}
   {% set hintText = "We will save your application." %}
-{% elif saveAndResume and (journey == "update") %}
+{% elif journey == "update" %}
```
Comment (markpit) [inline views/sign-out.html:21]: looks like the controllers/update/update.sign.out.controller.ts will still need to pass a saveAndResume value into this template. So i think we would need to revert this template and change the controllers/sign.out.controller.ts to pass saveAndResume: true into this template, until the update flags are removed.

File: src/controllers/update/update.resume.submission.controller.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mattch1) [reply src/controllers/update/update.resume.submission.controller.ts]: Reverted this work, just passing in true for resume submission.

File: test/controllers/payment.controller.spec.ts (hunk 2 above)
Comment (mattch1) [reply test/controllers/payment.controller.spec.ts]: Removed this.

File: test/controllers/sign.out.controller.spec.ts (hunk 3 above)
Comment (mattch1) [reply test/controllers/sign.out.controller.spec.ts:42]: Removed this.

File: views/sign-out.html (hunk 6 above)
Comment (mattch1) [reply views/sign-out.html]: Back out this change as update save and resume is still being used.

File: test/utils/save.and.continue.spec.ts (hunk 5 above)
Comment (mattch1) [reply test/utils/save.and.continue.spec.ts]: Kept and added a new test

File: src/controllers/sign.out.controller.ts [hunk 7]
```diff
@@ -14,7 +13,7 @@ export const get = (req: Request, res: Response, next: NextFunction) => {
     return res.render(config.SIGN_OUT_PAGE, {
       previousPage: previousPageUrl,
       url: config.REGISTER_AN_OVERSEAS_ENTITY_URL,
-      saveAndResume: isActiveFeature(config.FEATURE_FLAG_ENABLE_SAVE_AND_RESUME_17102022),
+      saveAndResume: true,
```
Comment (mwestacott) [inline src/controllers/sign.out.controller.ts:16]: Guessing we can't remove this (and the other associated things I've seen) until the Update save & resume flag is removed?

Comment (mattch1) [reply src/controllers/sign.out.controller.ts:16]: Yes, we can't remove this yet because of this update flag.

File: views/sign-out.html [hunk 8]
```diff
@@ -50,64 +51,63 @@
...
+         text: hintText
         },
         items: [
           {
             value: "yes",
             text: "Yes",
             attributes: {
-              "data-event-id": "yes-sign-out"
-            }
+            "data-event-id": "yes-sign-out"
```
Comment (markpit) [inline views/sign-out.html:61]: we've lost some of the indentation in this file, eg line 61, 68, 75-77 and a few more..
could you re-indent them and correct indention in the rest of this file? (views/sign-out.html)

File: src/middleware/service.availability.middleware.ts [hunk 9]
```diff
@@ -27,14 +24,5 @@ export const serviceAvailabilityMiddleware = (req: Request, res: Response, next:
...
 
-  if (
-    (req.path === STARTING_NEW_URL || req.path.endsWith(`/${RESUME}`))
-    &&
-    !isActiveFeature(FEATURE_FLAG_ENABLE_SAVE_AND_RESUME_17102022)) {
-    logger.infoRequest(req, "Save and resume flag is disabled - displaying service offline page");
-    return res.render(SERVICE_OFFLINE_PAGE);
-  }
-
-  // feature flag SHOW_SERVICE_OFFLINE_PAGE is false - carry on as normal
```
Comment (mwestacott) [inline src/middleware/service.availability.middleware.ts:38]: Strictly speaking, the comment on line 38 shouldn't be removed, although as it is not entirely accurate, this is probably fine. Alternatively, we reinstate it and correct it to be something like 'Service not offline, carry on as normal'. Not sure it adds much...

File: views/sign-out.html [hunk 10]
```diff
@@ -32,10 +32,11 @@
 
     <form method="post">
 
-      {# Need to have hidden fields for the journey and saveAndResume values so that when the template gets rendered by the validation middleware, #}
-      {#   the journey and saveAndResume values are not lost and passed back into the template for rendering #}
-      <input type="hidden" name="journey" value="{{ journey }}"/>
-      <input type="hidden" name="saveAndResume" value="{{ saveAndResume }}"/>
+      {# Need to have hidden fields for the journey and saveAndResume values so that when the template gets rendered by
+         the validation middleware, #}
```
Comment (mwestacott) [inline views/sign-out.html:36]: I think the content in line 36 should just be appended on to the end of the previous line, as it was before.

Comment (mattch1) [reply views/sign-out.html]: Tidied this.

File: src/middleware/service.availability.middleware.ts (hunk 9 above)
Comment (mattch1) [reply src/middleware/service.availability.middleware.ts:38]: Reintroduced and added detail to the comment

File: views/sign-out.html (hunk 8 above)
Comment (mattch1) [reply views/sign-out.html]: Indented this appropriately.

**Review by mwestacott — APPROVED** [review]

**Review by markpit — APPROVED** [review]

**Pull request merged**